plugins/modules/zos_blockinfile.py

```diff
@@ -666,19 +666,28 @@ def main():
 
     marker = "{0}\\n{1}\\n{2}".format(marker_begin, marker_end, marker)
     block = transformBlock(block, ' ', indentation)
+    gdg = False
     # analysis the file type
+    if "/" not in src:
+        dataset = data_set.MVSDataSet(
+            name=src
+        )
+        src = dataset.name
+        gdg = dataset.name
+
+    if data_set.DataSet.is_gds_relative_name(src) and gdg is False:
```

Comment:
I think I mentioned this in the previous review, but `gdg` variable name is used to store two different data types, and gdg variable is used for anything else than the negative case. What if we change this line to 

```suggestion
    elif data_set.DataSet.is_gds_relative_name(src):
```
and we just get rid of `gdg` variable here